broker/aws/api.go

```diff
@@ -4,6 +4,8 @@ import (
 	"sync"
 	"time"
 
+	"golang.org/x/sync/semaphore"
```

Comment:
You can also use a channel as a semaphore. It's easy to do, and it would avoid another dependency.

Reply:
I am fine with dependencies on golang.org/x.... and while we can use a channel as a sempahore I think it is less clear. I you really want we can move to channels, but I think we lose clarity. If you still want I can move to channel (with interface{} as what to send?)

Reply:
did the conversion to channel and is much less clear:
-       ctx := context.TODO()
-       if err := b.listRolesSemaphore.Acquire(ctx, 1); err != nil {
-               b.logger.Printf("Failed to acquire semaphore: %v", err)
-               return nil, err
-       }
-       defer b.listRolesSemaphore.Release(1)
+       b.listRolesSemaphore <- 1
+       defer func() { <-b.listRolesSemaphore }()

Yes it is smaller and one less dependency, but on clarity we fail. I think we should keep it as an explicit semaphore